### ISSUE TITLE
Should we add a lint rule, `avoid_positional_boolean_record_fields`?

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4125]
We have [avoid_positional_boolean_parameters](https://dart-lang.github.io/linter/lints/avoid_positional_boolean_parameters.html) backed by an Effective Dart rule. And I think the reasoning behind the rule would support a new `avoid_positional_boolean_record_fields` rule.

WDYT, @munificent ?

**Comment by munificent** [comment]
I think positional boolean record fields are an anti-pattern *in APIs*, but are surprisingly useful when you want to use ephemeral records to match over a combination of values:

```dart
for (var i = 1; i < 20; i++) {
  switch ((i % 3, i % 5)) {
    case (true, true): print('fizzbuzz');
    case (true, false): print('fizz');
    case (false, true): print('buzz');
    case (false, false): print(i);
  }
}
```

So I don't think we should generally discourage them. We can discourage them in APIs, but I'd like to see if that actually becomes a problem before we take up space in "Effective Dart" telling people not to do it.

**Comment by srawlins** [comment]
Ha that's a pretty good example.

OK we can also wait on a linter rule.
